src/Microsoft.AspNet.Razor.Runtime/BufferedTextWriter.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// TextWriter which is used to write <see cref="Runtime.BufferedStringCollection"/>.
+    /// </summary>
+    public class BufferedTextWriter : TextWriter
+    {
+        private static readonly Task _completedTask = Task.FromResult(0);
+        private readonly Encoding _encoding;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="BufferedTextWriter"/>.
+        /// </summary>
+        /// <param name="encoding">The character <see cref="Encoding"/> in which the output is written.</param>
```

Comment:
note: your `<see cref="Encoding" />` is referencing the property name and not the type name.